File: android/app/src/main/kotlin/com/verify/zsy/channels/ZsyEventChannel.java
package com.verify.zsy.channels;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.EventChannel;

public class ZsyEventChannel implements EventChannel.StreamHandler {
    private EventChannel channel;
    private EventChannel.EventSink eventSink;

    private ZsyEventChannel(BinaryMessenger binaryMessenger) {
        channel = new EventChannel(binaryMessenger, ZsyConstant.EVENT_CHANNEL);
        channel.setStreamHandler(this);
    }

    public static ZsyEventChannel create(BinaryMessenger binaryMessenger) {
        return new ZsyEventChannel(binaryMessenger);
    }

    @Override
    public void onListen(Object arguments, EventChannel.EventSink eventSink) {
        this.eventSink = eventSink;
        eventSink.success("this.arguments");
    }

    @Override
    public void onCancel(Object arguments) {
        this.eventSink = null;
    }

    protected void onCancel() {
        if (eventSink != null) {
            eventSink = null;
        }
        if (channel != null) {
            channel.setStreamHandler(null);
            channel = null;
        }
    }
}

File: android/app/src/main/kotlin/com/verify/zsy/ZsyEventChannel.kt
package com.verify.zsy

import android.app.Activity
import com.verify.zsy.ZsyConstant.EVENT_METHOD
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

/**
 * Android发送Flutter信息
 */
class ZsyEventChannel(private val activity: Activity, binaryMessenger: BinaryMessenger) : EventChannel.StreamHandler {

    private var eventChannel: EventChannel? = null
    private var eventSink: EventChannel.EventSink? = null

    init {
        eventChannel = EventChannel(binaryMessenger, EVENT_METHOD)
        eventChannel?.setStreamHandler(this)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        this.eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        this.eventSink = null
    }

    fun onCancel() {
        eventChannel?.setStreamHandler(null)
        eventChannel = null
    }
}